yarn-project/noir-libs/noir-aztec/src/note/note_getter_options.nr

```diff
@@ -29,50 +40,67 @@ impl Sort {
     }
 }
 
-struct NoteGetterOptions<Note, N, S, P> {
-    sort_by: [Sort; N],
+fn return_all_notes<Note, N>(notes: [Option<Note>; MAX_READ_REQUESTS_PER_CALL], _p: Field) -> [Option<Note>; MAX_READ_REQUESTS_PER_CALL] {
+    notes
+}
+
+// docs:start:NoteGetterOptions
+struct NoteGetterOptions<Note, N, FILTER_ARGS> {
+    selects: [Option<Select>; N],
+    sorts: [Option<Sort>; N],
+    limit: u32,
     offset: u32,
-    filter: fn ([Option<Note>; MAX_READ_REQUESTS_PER_CALL], P) -> [Option<Note>; S],
-    filter_args: P,
+    filter: fn ([Option<Note>; MAX_READ_REQUESTS_PER_CALL], FILTER_ARGS) -> [Option<Note>; MAX_READ_REQUESTS_PER_CALL],
+    filter_args: FILTER_ARGS,
 }
+// docs:end:NoteGetterOptions
 
-impl<Note, N, S, P> NoteGetterOptions<Note, N, S, P> {
-    fn new(
-        filter: fn ([Option<Note>; MAX_READ_REQUESTS_PER_CALL], P) -> [Option<Note>; S],
-        filter_args: P,
-        sort_by: [Sort; N],
-        offset: u32,
-    ) -> Self {
+impl<Note, N, FILTER_ARGS> NoteGetterOptions<Note, N, FILTER_ARGS> {
+    fn new() -> NoteGetterOptions<Note, N, Field> {
         NoteGetterOptions {
-            sort_by,
-            offset,
-            filter,
-            filter_args,
+            selects: [Option::none(); N],
+            sorts: [Option::none(); N],
+            limit: MAX_READ_REQUESTS_PER_CALL as u32,
+            offset: 0,
+            filter: return_all_notes,
+            filter_args: 0,
         }
     }
 
     fn with_filter(
-        filter: fn ([Option<Note>; MAX_READ_REQUESTS_PER_CALL], P) -> [Option<Note>; S],
-        filter_args: P,
+        filter: fn ([Option<Note>; MAX_READ_REQUESTS_PER_CALL], FILTER_ARGS) -> [Option<Note>; MAX_READ_REQUESTS_PER_CALL],
+        filter_args: FILTER_ARGS,
     ) -> Self {
          NoteGetterOptions {
-            sort_by: [Sort::nada(); N],
+            selects: [Option::none(); N],
+            sorts: [Option::none(); N],
+            limit: 0,
             offset: 0,
             filter,
             filter_args,
         }
     }
 
-    fn with_filter_and_offset(
-        filter: fn ([Option<Note>; MAX_READ_REQUESTS_PER_CALL], P) -> [Option<Note>; S],
-        filter_args: P,
-        offset: u32,
-    ) -> Self {
-         NoteGetterOptions {
-            sort_by: [Sort::nada(); N],
-            offset,
-            filter,
-            filter_args,
-        }
+    fn select(&mut self, field_index: u8, value: Field) -> Self {
```

Comment:
Perhaps we need block comments describing what each of these methods does, and how to use them? (Unless the docs will cover these in detail?)